documentation/development-docs/patterns-for-standard-cmdlets.md

```diff
@@ -0,0 +1,996 @@
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.IsParameterBound(c => c.InputObject))
+            {
+                this.ResourceGroupName = this.InputObject.ResourceGroupName;
+                this.Name = this.InputObject.Name;
+            }
+
+            if (this.IsParameterBound(c => c.ResourceId))
+            {
+                var resourceIdentifier = new ResourceIdentifier(this.ResourceId);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.Name = resourceIdentifier.ResourceName;
+            }
+
+            if (this.ShouldProcess(this.Name, string.Format("Deleting Foo '{0}' in resource group {0}", this.Name, this.ResourceGroupName)))
+            {
+                this.MySDKClient.Foo.Delete(this.ResourceGroupName, this.Name);
+                if (this.IsPassThru.IsPresent)
+                {
+                    WriteObject(true);
+                }
+            }
+        }
+    }
+}
+```
+
+</p>
+</details>
+
+### `Set-*` cmdlet
+
+All top-level resources should have a `Set-*` cmdlet that allows users to update an existing resource _if the API follows `PUT` semantics_. If the API supports `PATCH` semantics, then the cmdlet should be `Update-*` (see below). The user can update an existing resource by providing all identity properties, the resource id, or the object representation of the resource. This cmdlet should implement `SupportsShouldProcess` to allow users to provide the `-WhatIf` parameter and see what the result of executing the cmdlet is without the resource actually being updated.
+
+#### Parameter Sets
+
+To enable the scenarios mentioned previously, the cmdlet will need three parameter sets:
+
+```
+Set-AzureRmFoo -ResourceGroupName <String> -Name <String> -Property1 <Type1> -Property2 <Type2> ... [-WhatIf] [-Confirm]
+
+Set-AzureRmFoo -InputObject <PSFoo> [-Property1 <Type1>] -[Property2 <Type2>] ... [-WhatIf] [-Confirm]
+
+Set-AzureRmFoo -ResourceId <String> -Property1 <Type1> -Property2 <Type2> ... [-WhatIf] [-Confirm]
+```
+
+The first parameter set has required `-ResourceGroupName` and `-Name` parameters, as well as required property parameters to set their values on the resource. The second parameter has a required `-InputObject` parameter, as well as optional property parameters that override the value of the property on the given object if provided. The third parameter has a required `-ResourceId` parameter, as well as required property parameters to set their values on the resource.
+
+#### C# example
+
+<details><summary>Click to expand example</summary>
+<p>
+
+```cs
+namespace Microsoft.Azure.Commands.Foo
+{
+    [Cmdlet(VerbsCommon.Set, "AzureRmFoo", DefaultParameterSet = "UpdateByFooNameParameterSet", SupportsShouldProcess = true), OutputType(typeof(PSFoo))]
+    public class SetFooCommand : FooBaseCmdlet
+    {
+        [Parameter(Mandatory = true, ParameterSetName = "UpdateByFooNameParameterSet")]
```

Comment:
I would use 'SetBy' for parameter set names to avoid confusion with Update cmdlets

Reply:
Done.